### ISSUE TITLE
enum member key should reject bigint

### ISSUE THREAD

**Issue by JLHwung** [microsoft/TypeScript#61772]
### 🔎 Search Terms

enum, bigint, member key

### 🕗 Version & Regression Information

This is the behavior in every version I tried.

### ⏯ Playground Link

https://www.typescriptlang.org/play/?#code/KYOwrgtgBAolDeAoKKoAYRQLzsQXyA

### 💻 Code

```ts
enum E {
    0n = 0
}
```

### 🙁 Actual behavior

Transformed to
```js
"use strict";
var E;
(function (E) {
    E[E[0n] = 0] = 0n;
})(E || (E = {}));

```
without error

### 🙂 Expected behavior

It should throw an error "An enum member cannot have a numeric name.", similar to how we handle `enum E { 0 = 0 }`. Disallowing numeric value `0` but allowing bigInt value `0n` as enum member key seems inconsistent.

### Additional information about the issue

First reported in https://github.com/babel/babel/issues/17331.
